hoomd/write/table.py

```diff
@@ -208,12 +208,16 @@ def __init__(self,
 
         # internal variables that are not part of the state.
         # Ensure that only scalar and potentially string are set for the logger
-        if (LoggerCategories.scalar not in logger.categories
-                or logger.categories & self._invalid_logger_categories
-                !=  # noqa: W504 (yapf formats this incorrectly
-                LoggerCategories.NONE):
+        if LoggerCategories.scalar not in logger.categories:
+            raise ValueError("Given Logger must have the scalar categories set.")
+        elif (
+            logger.categories & self._invalid_logger_categories != LoggerCategories.NONE
+        ):
```

Comment:
Yes, as far as I am aware: the cases are as follows.

**If scalar category is not set:** the logger fails. This is in discussion [below](https://github.com/glotzerlab/hoomd-blue/pull/1519#discussion_r1138716821), but would work the same way if the case checked for scalar OR string

**If there are logger categories AND invalid logger categories:** the logger fails as intended.

**Other cases:**

- **There are no logger categories:** logger passes, but provides no output (expected)
- **There are logger categories but no invalid categories:** logger passes.
- **There are ONLY invalid categories:** this is the same as the first case, and will fail because there is no scalar category set.
- **There is only the string category:** this is also the same as the first case, and fails because scalar is not set

Reply:
It would be awesome if you added a pytest unit test that checks these.

Reply:
Added - see [68df2a](https://github.com/glotzerlab/hoomd-blue/pull/1519/commits/68df2a30053109ba1eef7edd49e10a6cfcc373a4)

Reply:
Thanks!